main.go

```diff
@@ -105,12 +107,17 @@ func main() {
 		}
 	}
 
-	factory := crdinformers.NewSharedInformerFactory(
+	var factoryOpts []crdinformers.SharedInformerOption
+	if *namespace != "" {
```

Comment:
I think we can be more defensive by doing `if namespace != nil && *namespace != ""`.

Reply:
ah good call